src/plugins/intel_gna/src/transformations/rt_info/gather_sinking_attr.hpp

```diff
@@ -0,0 +1,33 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/core/node.hpp"
+#include "openvino/core/runtime_attribute.hpp"
+#include "transformations_visibility.hpp"
+
+namespace ov {
+
+TRANSFORMATIONS_API void mark_as_no_gather_sinking_node(const std::shared_ptr<Node>& node);
+
+TRANSFORMATIONS_API bool is_gather_sinking_node(const std::shared_ptr<Node>& node);
+TRANSFORMATIONS_API bool is_gather_sinking_node(const Node* node);
+TRANSFORMATIONS_API bool is_gather_sinking_node(ov::Output<ov::Node> output);
+
+/**
+ * @ingroup ie_runtime_attr_api
+ * @brief NoGatherSinkingAttr class represents runtime info attribute that marks gather
+ * operation should not be moved be backward sinking propagation.
+ */
+class TRANSFORMATIONS_API NoGatherSinkingAttr : public RuntimeAttribute {
```

Comment:
I am new to this kind of transformation and it is not clear for me.
Are they automatically registered or they are currently used only in tests.